h2o-py/h2o/frame.py

```diff
@@ -448,15 +448,17 @@ def show(self, use_pandas=False, rows=10, cols=200):
             print("This H2OFrame is empty.")
             return
         if not self._ex._cache.is_valid(): self._frame()._ex._cache.fill()
-        if H2ODisplay._in_ipy():
+        if H2ODisplay._in_zep():
+            print("%html " + self._ex._cache._tabulate("html", False, rows=rows))
+        elif H2ODisplay._in_ipy():
             import IPython.display
             if use_pandas and can_use_pandas():
-                IPython.display.display(self.head(rows=rows, cols=cols).as_data_frame(fill_cache=True))
+                IPython.display.display(self.head(rows=rows, cols=cols).as_data_frame(use_pandas=True))
             else:
-                IPython.display.display_html(self._ex._cache._tabulate("html", False), raw=True)
+                IPython.display.display_html(self._ex._cache._tabulate("html", False, rows=rows), raw=True)
         else:
             if use_pandas and can_use_pandas():
-                print(self.head(rows=rows, cols=cols).as_data_frame(True))  # no keyword fill_cache
+                print(self.head(rows=rows, cols=cols).as_data_frame(use_pandas=True))
```

Comment:
I can see the `if-else` checks for zeppelin & Jupyter more times in the code. What do you think of creating a method like `print/show/display` in `display.py` file which would handle this internally and the rest of the code would just call this method?

Reply:
everywhere where it is possible we use H2ODisplay directly, the two frame method have special behavior where it is best to keep it like this.
It would be possible to rewrite the code to move most of the logic to H2ODisplay but it would be a major project with little benefit

Reply:
Thanks for the explanation!